quantlib/instruments/bonds.pyx

```diff
@@ -212,3 +216,53 @@ cdef class ZeroCouponBond(Bond):
                 'Wrapper for such constructor not yet implemented.'
             )
 
+cdef class FloatingRateBond(Bond): 
+    """ Floating Rate Bond """
+    def __init__(self, int settlement_days, double face_amount, Schedule float_schedule, 
+        IborIndex ibor_index, DayCounter accrual_day_counter, int fixing_days, 
+        gearings, spreads, caps, floors, payment_convention=Following, redemption=100.0, Date issue_date=None
+        ):
+        """Floating rate bond (Instantiation): 
+            1. (int) settlement_days
+            2. (double) face_amount
+            3. (Quantlib::Schedule) 
+            4. (Quantlib::IborIndex)
+            5. (Quantlib::DayCounter)
+            6. (int) fixing_days
+            7. (List[double]) gearings
+            8. (List[Quantlib::Spread]) spreads
+            9. (List[double]) caps
+            10. (List[double]) floors
+            11. (Quantlib::BusinessDayConvention)
+            12. (double) redemption
+            13. (Quantlib::Date) issue_date
+          """
+        cdef QlSchedule* _float_bonds_schedule = <QlSchedule*>float_schedule._thisptr
+        cdef QlDayCounter* _accrual_day_counter = <QlDayCounter*>accrual_day_counter._thisptr
+        cdef _date.Date* _issue_date
+        
+        
+        cdef vector[Real] _gearings = vector[Real]()
+        cdef vector[Spread] _spreads = vector[Spread]()
+        cdef vector[Rate] _caps = vector[Rate]()
+        cdef vector[Rate] _floors = vector[Rate]()
+        
+        for item in gearings: 
+            _gearings.push_back(item)
+        for spd in spreads: 
+            _spreads.push_back(spd)
+        for rtc in caps:
+            _caps.push_back(rtc)
+        for rtf in floors: 
+            _floors.push_back(rtf)
+        
+        _issue_date = <_date.Date*>((<Date>issue_date)._thisptr.get())
```

Comment:
do you really need that explicity `<data.Date*> cast? `.get()` is supposed to return a `T*` and thus the appropriate pointer.